frontend/src/views/GameView.vue

```diff
@@ -1,51 +1,156 @@
+					</div>
+
+					<canvas
+					ref="GameRef"
+					id="GameCanvas"
+					class="gameBlock"
+					width="1200"  
+					height="960">
+					</canvas>
+
+				</div>
 		</div>
-		<p class="table table-striped table-sm">
-			All Games:
-			<tbody>
-				<tr v-for="game in gameStore.allGames" :key="game.id">
-					<td> Game id: {{ game.id }} </td>
-					<td> Score player one: {{ game.score_player_one }} </td>
-					<td> Score player two: {{ game.score_player_two }} </td>
-					<td>Is active: {{ game.is_active }}</td>
-				</tr>
-				
-			</tbody>
-
-		</p>
-		<p class="table table-striped table-sm">
-			All My Games:
-			<tbody>
-				<tr v-for="game in gameStore.getMyGames" :key="game.id">
-					<td>Game id: {{ game.id }}</td>
-					<td>Score player one: {{ game.score_player_one }}</td>
-					<td>Score player two: {{ game.score_player_two }}</td>
-					<td>Is active: {{ game.is_active }}</td>
-				</tr>
-				
-			</tbody>
-			
-		</p>
 	</div>
 </template>
 
-<!--  THIS SCRIPT IS ONLY TO SHOW A WORKING GAME STORE SETUP-->
 <script lang="ts">
 import { useGameStore } from '@/stores/gameStore';
 import { defineComponent } from 'vue';
+import PlayerNames from '@/components/game-info/PlayerNames.vue'
+
+interface DataObject {
+	context: CanvasRenderingContext2D
+}
 
 export default defineComponent({
 	name: "GameView",
+	components: {
+		PlayerNames
+	},
+
+	data(): DataObject {
+		return {
+			context: {} as CanvasRenderingContext2D
+		}
+	},
 
 	setup(){
 		const gameStore = useGameStore();
 		gameStore.refreshData();
 		return {
 			gameStore,
 		}
-	}
+	},
+	computed: {
+		// typescript wants values used from html to have a type
+		width() {
+			return (this.$refs.GameRef as HTMLCanvasElement).width;
+		},
+		height() {
+			return (this.$refs.GameRef as HTMLCanvasElement).height;
+		},
+	},
+	methods: {
+		// https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/fillRect
+		drawMiddleLine() {
+			const widthStart: number = this.width / 2;
+			const heightStart: number = 0;
+			const lineWidth: number = 4;
+			const lineHeight: number = this.height; 
+			
+			this.context.fillRect(
+					widthStart,
+					heightStart,
+					lineWidth,
+					lineHeight
+				);
+			},
+			
+		//  https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/arc
+		drawBall() {
+			const startingX: number = this.width / 2 - 100;
+			const startingY: number = this.height / 2 - 100;
+			const radius: number = 15;
+			const startAngle: number = 0;
+			const endAngle: number = Math.PI * 2; // full circle
+
+			this.context.arc(
+					startingX,
+					startingY,
+					radius,
+					startAngle,
+					endAngle,
+			)
+				// this fills the above defined arc/circle
+				this.context.fill()
+			},
+				
+		// https://developer.mozilla.org/en-US/docs/Web/API/CanvasRenderingContext2D/fillRect
+		drawPaddles() {
+			let widthStart: number = 0;
+			const lineWidth: number = 15;
+			const lineHeight: number = this.height / 5; 
```

Comment:
the lineHeight now becomes 20% of the canvas height.
Which is, to me, not the clearest way to calculate this (as in, this does not immediately shout "i am an percentage" to me (`this.height / 100 * 20` does that more, but is annoying)
so maybe we can add some computed property that is 1% of the width and another one that is 1% of the height? (just a suggestion) because then you can just always use that and multiply by the percentage wanted.


also:
I guess the paddle-height multiplier / divisor should be a variable as well (if we ever want to make the paddles change size when you make a point or something like that..) Not sure if we should already consider different paddle heights per player.......................

Reply:
changed to percentage values.

Added types for later use. All moving elements will be variables, that only starts making sense when movement starts as now it will be a declaration and then filling in values, cluttering unnecessarily . For now this seems clearer as it's just const values being drawn once.

Reply:
good points :) const values for now are fine!